opentelemetry-exporter-geneva/geneva-uploader/src/config_service/client.rs

```diff
@@ -370,10 +372,23 @@ impl GenevaConfigClient {
                 }
             }
         }
-        // Cache miss or expired token, fetch fresh data
-        // Perform actual fetch before acquiring write lock to minimize lock contention
-        let (fresh_ingestion_gateway_info, fresh_moniker_info) =
-            self.fetch_ingestion_info().await?;
+        // Cache miss or expired token, fetch fresh data with retry logic (up to 5 retries on retriable errors)
+        let (fresh_ingestion_gateway_info, fresh_moniker_info) = {
+            let mut attempt = 0;
+            loop {
+                match self.fetch_ingestion_info().await {
+                    Ok(v) => break v,
+                    Err(e) => {
+                        if attempt < MAX_GCS_RETRIES && is_retriable_error(&e) {
+                            attempt += 1;
+                            continue;
+                        } else {
```

Comment:
We need to respect the [`Retry-After`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Retry-After) header if present, before sending the request again.